timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -529,14 +588,15 @@ void __global__ k_nonbonded_unified(
     const double cutoff,
     const int *__restrict__ ixn_tiles,
     const unsigned int *__restrict__ ixn_atoms,
+    const unsigned int row_atom_offset, // index of the first atom in the first row block
     unsigned long long *__restrict__ du_dx,
     unsigned long long *__restrict__ du_dp,
     unsigned long long *__restrict__ du_dl_buffer,
     unsigned long long *__restrict__ u_buffer) {
 
     int tile_idx = blockIdx.x;
     int row_block_idx = ixn_tiles[tile_idx];
-    int atom_i_idx = row_block_idx * 32 + threadIdx.x;
+    int atom_i_idx = row_block_idx * 32 + threadIdx.x + row_atom_offset;
```

Comment:
still a bit unsure about this. The idea is to make `atom_i_idx` index into the full permuted coordinate array of size `N1_ + N2_`